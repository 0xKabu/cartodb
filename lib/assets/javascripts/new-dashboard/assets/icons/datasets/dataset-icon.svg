<?xml version="1.0" encoding="UTF-8"?>
<svg width="36px" height="40px" viewBox="0 0 36 40" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 52.6 (67491) - http://www.bohemiancoding.com/sketch -->
    <title>server-2</title>
    <desc>Created with Sketch.</desc>
    <g id="Active-Development-Cycle" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="01-Home-/-Recent" transform="translate(-382.000000, -318.000000)">
            <g id="Group-6" transform="translate(376.000000, 312.000000)">
                <g id="places-home-1">
                    <g id="Frame_-_24px">
                        <g id="server-2" transform="translate(0.000000, 2.000000)">
                            <g id="Frame-24px">
                                <rect id="Rectangle" x="0" y="0" width="48" height="48"></rect>
                            </g>
                            <g id="Line_Icons" transform="translate(6.000000, 4.000000)" fill="#FFFFFF" fill-rule="nonzero">
                                <path d="M18,0 C9.054,0 0,2.748 0,8 L0,32 C0,37.252 9.054,40 18,40 C26.946,40 36,37.252 36,32 L36,8 C36,2.748 26.946,0 18,0 Z M18,4 C27.094,4 32,6.84 32,8 C32,9.16 27.094,12 18,12 C8.906,12 4,9.16 4,8 C4,6.84 8.906,4 18,4 Z M18,36 C8.906,36 4,33.16 4,32 L4,29.2 C7.472,31.046 12.748,32 18,32 C23.252,32 28.528,31.044 32,29.2 L32,32 C32,33.16 27.094,36 18,36 Z M18,28 C8.906,28 4,25.16 4,24 L4,21.2 C7.472,23.046 12.748,24 18,24 C23.252,24 28.528,23.044 32,21.2 L32,24 C32,25.16 27.094,28 18,28 Z M18,20 C8.906,20 4,17.16 4,16 L4,13.2 C7.472,15.044 12.748,16 18,16 C23.252,16 28.528,15.044 32,13.2 L32,16 C32,17.16 27.094,20 18,20 Z" id="Shape"></path>
                            </g>
                        </g>
                    </g>
                </g>
            </g>
        </g>
    </g>
</svg>