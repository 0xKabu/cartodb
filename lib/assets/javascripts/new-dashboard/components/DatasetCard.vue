<template>
  <a :href="dataset.url" class="row" :class="{'selected': selected, 'card--noHover': !activeHover}">
    <div class="cell cell--start">
      <div class="row-dataType">
          <div class="icon--dataType" :class="'icon--' + dataType"></div>
        <!-- <img src="../assets/icons/datasets/data-types/{{dataType}}.svg"> -->
        <!-- {%include img/icons/rows/data-types/dots.svg%} -->
      </div>
      <span class="checkbox row-checkbox" style="margin-right: 20px;">
        <input class="checkbox-input" @click="toggleSelection" type="checkBox" name="contact" value="02">
        <span class="checkbox-decoration">
          <img svg-inline src="../assets/icons/common/checkbox.svg">
        </span>
      </span>
    </div>
    <div class="cell cell--main">
      <h3 class="text is-caption is-txtGrey u-ellipsis row-title">
        {{ dataset.name }}
      </h3>
      <div class="row-metadataContainer">
        <div class="row-metadata" v-if="numberTags > 0">
          <!-- {%include img/icons/rows/tag.svg%} -->
          <img class="icon-metadata" svg-inline src="../assets/icons/datasets/tag.svg">
          <ul class="tag-list">
            <li v-for="(tag, index) in dataset.tags" :key="tag">
              <a href="#" class="text is-small is-txtSoftGrey">{{ tag }}<span v-if="index < numberTags - 1">,&nbsp;</span></a>
            </li>
          </ul>
          <!-- <span class="text is-small is-txtBaseGrey">spain, 2018</span> -->
        </div>
        <div class="row-metadata" v-if="isShared">
          <img class="icon-metadata" svg-inline src="../assets/icons/datasets/user.svg">
          <!-- {%include img/icons/rows/user.svg%} -->
          <span class="text is-small is-txtSoftGrey">Timchung</span>
        </div>
      </div>
    </div>
    <div class="cell cell--large">
      <span class="text is-small is-txtSoftGrey">{{ lastSynced }}</span>
    </div>
    <div class="cell cell--small">
      <span class="text is-small is-txtSoftGrey">{{ numberRows }}</span>
    </div>
    <div class="cell cell--small">
      <span class="text is-small is-txtSoftGrey">{{ humanFileSize(dataset.table.size) }}</span>
    </div>
    <div class="cell cell--small">
      <span class="text is-small is-txtSoftGrey">1 map</span>
    </div>
    <div class="cell cell--small cell--privacy">
      <span class="icon icon--privacy" :class="privacyIcon"></span>
      <span class="text is-small is-txtSoftGrey">{{ $t(`DatasetCard.shared.${dataset.privacy}`) }}</span>
    </div>
    <div class="cell">
      <!-- QUICK ACTIONS -->
    </div>
  </a>
</template>

<script>
import distanceInWordsStrict from 'date-fns/distance_in_words_strict';
import * as Visualization from 'new-dashboard/core/visualization';

export default {
  name: 'DatasetCard',
  props: {
    dataset: Object
  },
  data: function () {
    return {
      selected: false,
      activeHover: true
    };
  },
  computed: {
    privacyIcon () {
      return `icon--${this.$props.dataset.privacy}`.toLowerCase();
    },
    lastSynced () {
      return this.$t(`DatasetCard.lastSynced`, { date: distanceInWordsStrict(this.$props.dataset.updated_at, new Date()) });
    },
    numberRows () {
      if (this.$props.dataset.table.row_count === 1) {
        return this.$t(`DatasetCard.numberRow`, { rows: this.$props.dataset.table.row_count });
      } else {
        return this.$t(`DatasetCard.numberRows`, { rows: this.$props.dataset.table.row_count });
      }
    },
    dataType () {
      let data = this.$props.dataset.table.geometry_types[0];
      return data ? data.replace('ST_', '').toLowerCase() : '';
    },
    numberTags () {
      return this.$props.dataset.tags ? this.$props.dataset.tags.length : 0;
    },
    isShared () {
      return Visualization.isShared(this.$props.dataset, this.$cartoModels);
      // return false;
    }
  },
  methods: {
    toggleSelection () {
      this.selected = !this.selected;
    },
    humanFileSize (size) {
      var i = Math.floor(Math.log(size) / Math.log(1024));
      return (size / Math.pow(1024, i)).toFixed(2) * 1 + ' ' + ['B', 'kB', 'MB', 'GB', 'TB'][i];
    }
  }
};
</script>

<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.row {
  display: flex;
  align-items: center;
  width: 100%;
  height: 80px;
  padding: 12px 14px;
  overflow: hidden;
  border-bottom: 1px solid $light-grey;
  background-color: $white;

  &.selected {
    background-color: $softblue;
  }

  &.selected,
  &:hover {
    text-decoration: none;

    .row-dataType {
      transform: translateY(-100%);
      opacity: 0;
    }

    .row-checkbox {
      transform: translateY(0);
      opacity: 1;
      pointer-events: all;
    }
  }

  &:hover {
    .row-title {
      color: $primary-color;
    }
  }
}

.tag-list {
  display: flex;
}

.row-dataType {
  width: 36px;
  height: 36px;
  padding: 9px;
  overflow: hidden;
  transition: all 0.25s cubic-bezier(0.4, 0.01, 0.165, 0.99);
  border-radius: 4px;
  background-color: $light-grey;

  .row-typeIcon {
    fill: $text-color;
  }

  &.row-dataType--premium {
    background-color: $premium-color;

    .row-typeIcon {
      fill: $white;
    }
  }
}

.row-checkbox {
  position: absolute;
  // top: 6px;
  left: 6px;
  transform: translateY(250%);
  transition: all 0.25s cubic-bezier(0.4, 0.01, 0.165, 0.99);
  opacity: 0;
  pointer-events: none;
}

.row-metadataContainer {
  display: flex;
  align-items: center;
  margin-top: 4px;
}

.row-metadata {
  display: flex;
  align-items: center;
  margin-left: 16px;

  &:first-of-type {
    margin-left: 0;
  }
}

.cell {
  position: relative;
  flex-grow: 0;
  flex-shrink: 0;
  padding: 0 10px;
  text-overflow: ellipsis;
  white-space: nowrap;

  &:first-of-type {
    padding-left: 0;
  }

  &:last-of-type {
    padding-right: 0;
  }
}

.cell--start {
  display: flex;
  align-items: center;
  align-self: flex-start;
  height: 100%;
}

.cell--main {
  flex-grow: 1;
  flex-shrink: 1;
  min-width: 200px;
}

.cell--large {
  width: 165px;
}

.cell--medium {
  width: 120px;
}

.cell--small {
  width: 80px;
}

.cell--privacy {
  display: flex;
  align-items: center;
}

.icon--privacy {
  display: flex;
  justify-content: center;
  width: 18px;
  height: 18px;
  margin-right: 8px;
  background-repeat: no-repeat;
  background-position: center;

  &.icon--private {
    background-image: url("../assets/icons/datasets/privacy/lock.svg");
  }

  &.icon--public {
    background-image: url("../assets/icons/datasets/privacy/public.svg");
  }

  &.icon--link {
    background-image: url("../assets/icons/datasets/privacy/link.svg");
  }

  &.icon--password {
    background-image: url("../assets/icons/datasets/privacy/password.svg");
  }

  &.icon--sharedBy {
    border-radius: 2px;
    background-size: contain;
  }
}

.icon--dataType {
  width: 100%;
  height: 100%;
  background-repeat: no-repeat;
  background-size: cover;

  &.icon--point {
    background-image: url("../assets/icons/datasets/data-types/dots.svg");
  }

  &.icon--multipolygon {
    background-image: url("../assets/icons/datasets/data-types/area.svg");
  }
}

.icon-metadata {
  margin-right: 4px;
}
</style>
