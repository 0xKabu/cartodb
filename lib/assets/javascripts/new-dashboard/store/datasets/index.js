import CartoNode from 'carto-node';
import toObject from 'new-dashboard/utils/to-object';
import Filters, { defaultParams as filtersDefaultParams } from 'new-dashboard/core/filters';

const client = new CartoNode.AuthenticatedClient();

const datasets = {
  namespaced: true,
  state: {
    isFetching: false,
    isFiltered: false,
    isErrored: false,
    error: {},
    filterType: 'mine',
    order: 'updated_at',
    list: {},
    metadata: {},
    page: 1,
    numPages: 1
  },
  mutations: {
    setRequestError (state, error) {
      state.isFetching = false;
      state.isErrored = true;
      state.error = error;
    },

    setDatasets (state, datasets) {
      state.list = toObject(datasets.visualizations, 'id');
      state.metadata = {
        total_entries: datasets.total_entries,
        total_likes: datasets.total_likes,
        total_shared: datasets.total_shared,
        total_user_entries: datasets.total_user_entries
      };

      state.isFetching = false;
    },

    setFetchingState (state) {
      state.isFetching = true;
      state.isErrored = false;
      state.error = {};
    },

    setURLOptions (state, options) {
      state.page = options.page ? parseInt(options.page) : 1;
      state.filterType = options.filter || 'mine';
    },

    setPagination (state, page) {
      state.page = page;
      state.numPages = Math.ceil(state.metadata.total_entries / filtersDefaultParams.per_page) || 1;
    }
  },
  actions: {
    fetchDatasets (context) {
      context.commit('setFetchingState');

      const params = {
        ...Filters[context.state.filterType],
        types: 'table',
        page: context.state.page,
        order: context.state.order
      };

      client.getVisualization('',
        params,
        function (err, _, data) {
          if (err) {
            context.commit('setRequestError', err);
            return;
          }
          context.commit('setDatasets', data);
          context.commit('setPagination', context.state.page);
        });
    },

    setURLOptions (context, options) {
      context.commit('setURLOptions', options);
      context.dispatch('fetchDatasets');
    }
  }
};

export default datasets;
