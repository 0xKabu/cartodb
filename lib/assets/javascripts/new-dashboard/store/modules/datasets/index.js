import Filters from 'new-dashboard/core/configuration/filters';

import * as SharedActions from '../../actions/shared';
import * as SharedMutations from '../../mutations/shared';
import * as VisualizationActions from '../../actions/visualizations';
import * as VisualizationMutations from '../../mutations/visualizations';

const DEFAULT_VALUES = {
  filter: 'mine',
  order: 'favorited,updated_at',
  orderDirection: 'desc,desc'
};

const datasets = {
  namespaced: true,
  state: {
    isFetching: false,
    isFiltered: false,
    isErrored: false,
    error: {},
    filterType: DEFAULT_VALUES.filter,
    order: DEFAULT_VALUES.order,
    orderDirection: DEFAULT_VALUES.orderDirection,
    list: {},
    metadata: {},
    page: 1,
    numPages: 1,
    resultsPerPage: 1
  },
  mutations: {
    setFetchingState: SharedMutations.setFetchingState,
    setFilterType: VisualizationMutations.setFilterType,
    setOrder: VisualizationMutations.setOrder(DEFAULT_VALUES),
    setPagination: SharedMutations.setPagination,
    setResultsPerPage: SharedMutations.setResultsPerPage,
    setRequestError: SharedMutations.setRequestError,
    setVisualizations: VisualizationMutations.setVisualizations,
    updateNumberLikes: VisualizationMutations.updateNumberLikes,
    updateVisualization: VisualizationMutations.updateVisualization
  },
  actions: {
    deleteLike: VisualizationActions.deleteLike,
    filter: VisualizationActions.filter,
    like: VisualizationActions.like,
    order: VisualizationActions.order,
    resetFilters: VisualizationActions.resetFilters(DEFAULT_VALUES),
    setResultsPerPage: SharedActions.setResultsPerPage,
    setURLOptions: VisualizationActions.setURLOptions,
    updateVisualization: VisualizationActions.updateVisualization,

    fetch (context) {
      const params = {
        ...Filters[context.state.filterType],
        types: 'table',
        page: context.state.page,
        order: context.state.order,
        order_direction: context.state.orderDirection,
        per_page: context.state.resultsPerPage,
        with_dependent_visualizations: 10
      };

      VisualizationActions.fetchVisualizations(context, params);
    }
  }
};

export default datasets;
