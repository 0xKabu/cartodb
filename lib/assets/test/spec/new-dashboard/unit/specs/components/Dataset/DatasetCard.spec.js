import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import datasets from '../../fixtures/datasets';
import usersArray from '../../fixtures/users';
import DatasetCard from 'new-dashboard/components/Dataset/DatasetCard';

// Backbone Models
import ConfigModel from 'dashboard/data/config-model';
import UserModel from 'dashboard/data/user-model';

const localVue = createLocalVue();
localVue.use(Vuex);

let $cartoModels;
const $t = key => key;

function configCartoModels (attributes = {}) {
  const user = new UserModel(attributes.user || usersArray[0]);
  const config = new ConfigModel(attributes.config || { maps_api_template: 'http://{user}.example.com' });
  return { user, config };
}

describe('DatasetCard.vue', () => {
  beforeEach(() => {
    $cartoModels = configCartoModels();
  });

  it('should render correct contents', () => {
    const dataset = datasets.visualizations[0];
    const datasetCard = shallowMount(DatasetCard, {
      propsData: { dataset },
      mocks: {
        $cartoModels,
        $t
      }
    });

    expect(datasetCard).toMatchSnapshot();
  });

  it('should be able to select the card', () => {
    const dataset = datasets.visualizations[0];
    const datasetCard = shallowMount(DatasetCard, {
      propsData: { dataset },
      mocks: {
        $cartoModels,
        $t
      }
    });
    const checkbox = datasetCard.find('.checkbox-input');

    checkbox.trigger('click');

    expect(datasetCard).toMatchSnapshot();
  });

  it('should show the map is shared', () => {
    $cartoModels = configCartoModels({ user: usersArray[1] });
    const dataset = datasets.visualizations[0];
    const datasetCard = shallowMount(DatasetCard, {
      propsData: { dataset },
      mocks: {
        $cartoModels,
        $t
      }
    });

    expect(datasetCard).toMatchSnapshot();
  });

  it('should render tags', () => {
    const dataset = {...datasets.visualizations[0]};
    dataset.tags = ['Hi', 'Hello', 'CARTO Rules'];
    const datasetCard = shallowMount(DatasetCard, {
      propsData: { dataset },
      mocks: {
        $cartoModels,
        $t
      }
    });

    expect(datasetCard).toMatchSnapshot();
  });

  describe('Computed', () => {
    it('should generate the correct icon class', () => {
      const dataset = {...datasets.visualizations[0]};
      dataset.privacy = 'PUBLIC';
      const datasetCard = shallowMount(DatasetCard, {
        propsData: { dataset },
        mocks: {
          $cartoModels,
          $t
        }
      });

      const iconClass = datasetCard.vm.privacyIcon;

      expect(iconClass).toBe('icon--public');
    });

    describe('numberRows', () => {
      it('If there are zero rows, should print the plural of "row"', () => {
        const dataset = {...datasets.visualizations[0]};
        dataset.table.row_count = 0;
        const datasetCard = shallowMount(DatasetCard, {
          propsData: { dataset },
          mocks: {
            $cartoModels,
            $t
          }
        });

        expect(datasetCard.vm.numberRows).toBe('DatasetCard.numberRows');
      });

      it('If there is only one row, should print the singular of "row"', () => {
        const dataset = {...datasets.visualizations[0]};
        dataset.table.row_count = 1;
        const datasetCard = shallowMount(DatasetCard, {
          propsData: { dataset },
          mocks: {
            $cartoModels,
            $t
          }
        });

        expect(datasetCard.vm.numberRows).toBe('DatasetCard.numberRow');
      });

      it('If there are multiple rows, should print the plural of "row"', () => {
        const dataset = {...datasets.visualizations[0]};
        dataset.table.row_count = 2;
        const datasetCard = shallowMount(DatasetCard, {
          propsData: { dataset },
          mocks: {
            $cartoModels,
            $t
          }
        });

        expect(datasetCard.vm.numberRows).toBe('DatasetCard.numberRows');
      });
    });

    it('It should return the correct dataType"', () => {
      const dataset = {...datasets.visualizations[0]};
      dataset.table.geometry_types = ['ST_Multipolygon', 'ST_Point'];
      const datasetCard = shallowMount(DatasetCard, {
        propsData: { dataset },
        mocks: {
          $cartoModels,
          $t
        }
      });

      expect(datasetCard.vm.dataType).toBe('multipolygon');
    });

    describe('numberTags', () => {
      it('Should return 0 when the tags are null', () => {
        const dataset = {...datasets.visualizations[0]};
        dataset.tags = null;
        const datasetCard = shallowMount(DatasetCard, {
          propsData: { dataset },
          mocks: {
            $cartoModels,
            $t
          }
        });

        expect(datasetCard.vm.numberTags).toBe(0);
      });

      it('Should return the number of thags when we have tags', () => {
        const dataset = {...datasets.visualizations[0]};
        dataset.tags = ['test 1', 'Another test'];
        const datasetCard = shallowMount(DatasetCard, {
          propsData: { dataset },
          mocks: {
            $cartoModels,
            $t
          }
        });

        expect(datasetCard.vm.numberTags).toBe(2);
      });
    });

    describe('isShared', () => {
      it('Should return true if the user is different than the owner of the map', () => {
        $cartoModels = configCartoModels({ user: usersArray[1] });
        const dataset = datasets.visualizations[0];
        const datasetCard = shallowMount(DatasetCard, {
          propsData: { dataset },
          mocks: {
            $cartoModels,
            $t
          }
        });

        expect(datasetCard.vm.isShared).toBe(true);
      });

      it('Should return false if the user is different than the owner of the map', () => {
        const dataset = datasets.visualizations[0];
        const datasetCard = shallowMount(DatasetCard, {
          propsData: { dataset },
          mocks: {
            $cartoModels,
            $t
          }
        });

        expect(datasetCard.vm.isShared).toBe(false);
      });
    });

    describe('methods', () => {
      it('should toggle selection', () => {
        const dataset = datasets.visualizations[0];
        const datasetCard = shallowMount(DatasetCard, {
          propsData: { dataset },
          mocks: {
            $cartoModels,
            $t
          }
        });
        expect(datasetCard.vm.selected).toBe(false);

        datasetCard.vm.toggleSelection();
        expect(datasetCard.vm.selected).toBe(true);

        datasetCard.vm.toggleSelection();
        expect(datasetCard.vm.selected).toBe(false);
      });

      it('should return human readable file size', () => {
        const dataset = {...datasets.visualizations[0]};
        const datasetCard = shallowMount(DatasetCard, {
          propsData: { dataset },
          mocks: {
            $cartoModels,
            $t
          }
        });

        expect(datasetCard.vm.humanFileSize(0)).toBe('0 B');
        expect(datasetCard.vm.humanFileSize(1)).toBe('1 B');
        expect(datasetCard.vm.humanFileSize(1024)).toBe('1 kB');
        expect(datasetCard.vm.humanFileSize((Math.pow(1024, 2) * 4) + (Math.pow(1024, 2) / 3))).toBe('4.33 MB');
        expect(datasetCard.vm.humanFileSize(Math.pow(1024, 3))).toBe('1 GB');
      });

      it('should put the active hover to true', () => {
        const dataset = datasets.visualizations[0];
        const datasetCard = shallowMount(DatasetCard, {
          propsData: { dataset },
          mocks: {
            $cartoModels,
            $t
          }
        });

        datasetCard.vm.mouseOverChildElement();

        expect(datasetCard.vm.activeHover).toBe(false);
      });

      it('should put the active hover to false', () => {
        const dataset = datasets.visualizations[0];
        const datasetCard = shallowMount(DatasetCard, {
          propsData: { dataset },
          mocks: {
            $cartoModels,
            $t
          }
        });

        datasetCard.vm.mouseOutChildElement();

        expect(datasetCard.vm.activeHover).toBe(true);
      });
    });
  });
});
