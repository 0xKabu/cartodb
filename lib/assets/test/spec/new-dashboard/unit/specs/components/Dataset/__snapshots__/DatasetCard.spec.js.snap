// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetCard.vue should render correct contents 1`] = `
<a href="http://example.com/tables/untitled_table_13" class="dataset-row">
  <div class="dataset-cell cell--start">
    <div class="row-dataType">
      <div class="icon--dataType icon--point"></div>
    </div> <span class="checkbox row-checkbox"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../../assets/icons/common/checkbox.svg"></span></span>
  </div>
  <div class="dataset-cell cell--main">
    <div class="title-container">
      <h3 class="text is-caption is-txtGrey u-ellipsis row-title">
        untitled_table_13
      </h3> <span class="card-favorite"><img svg-inline="" src="../../assets/icons/common/favorite.svg"></span>
    </div>
    <!---->
  </div>
  <div class="dataset-cell cell--large"><span class="text is-small is-txtSoftGrey">DatasetCard.lastUpdated</span></div>
  <div class="dataset-cell cell--small"><span class="text is-small is-txtSoftGrey">DatasetCard.numberRows</span></div>
  <div class="dataset-cell cell--small"><span class="text is-small is-txtSoftGrey">34.4 MB</span></div>
  <div class="dataset-cell cell--small"><span class="text is-small is-txtSoftGrey">1 map</span></div>
  <div class="dataset-cell cell--small cell--privacy"><span class="icon icon--privacy icon--private"></span> <span class="text is-small is-txtSoftGrey">DatasetCard.shared.PRIVATE</span></div>
  <div class="dataset-cell"></div>
</a>
`;

exports[`DatasetCard.vue should render multiple tags 1`] = `
<a href="http://example.com/tables/untitled_table_13" class="dataset-row">
  <div class="dataset-cell cell--start">
    <div class="row-dataType">
      <div class="icon--dataType icon--point"></div>
    </div> <span class="checkbox row-checkbox"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../../assets/icons/common/checkbox.svg"></span></span>
  </div>
  <div class="dataset-cell cell--main">
    <div class="title-container">
      <h3 class="text is-caption is-txtGrey u-ellipsis row-title">
        untitled_table_13
      </h3> <span class="card-favorite"><img svg-inline="" src="../../assets/icons/common/favorite.svg"></span>
    </div>
    <div class="row-metadataContainer">
      <div class="row-metadata"><img svg-inline="" src="../../assets/icons/datasets/tag.svg" class="icon-metadata">
        <!---->
        <featuresdropdown-stub list="Hi,Hello,CARTO Rules,Just one more tag here"><span class="feature-text text is-small is-txtSoftGrey">4 DatasetCard.tags</span></featuresdropdown-stub>
      </div>
      <!---->
    </div>
  </div>
  <div class="dataset-cell cell--large"><span class="text is-small is-txtSoftGrey">DatasetCard.lastUpdated</span></div>
  <div class="dataset-cell cell--small"><span class="text is-small is-txtSoftGrey">DatasetCard.numberRows</span></div>
  <div class="dataset-cell cell--small"><span class="text is-small is-txtSoftGrey">34.4 MB</span></div>
  <div class="dataset-cell cell--small"><span class="text is-small is-txtSoftGrey">1 map</span></div>
  <div class="dataset-cell cell--small cell--privacy"><span class="icon icon--privacy icon--private"></span> <span class="text is-small is-txtSoftGrey">DatasetCard.shared.PRIVATE</span></div>
  <div class="dataset-cell"></div>
</a>
`;

exports[`DatasetCard.vue should render tags 1`] = `
<a href="http://example.com/tables/untitled_table_13" class="dataset-row">
  <div class="dataset-cell cell--start">
    <div class="row-dataType">
      <div class="icon--dataType icon--point"></div>
    </div> <span class="checkbox row-checkbox"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../../assets/icons/common/checkbox.svg"></span></span>
  </div>
  <div class="dataset-cell cell--main">
    <div class="title-container">
      <h3 class="text is-caption is-txtGrey u-ellipsis row-title">
        untitled_table_13
      </h3> <span class="card-favorite"><img svg-inline="" src="../../assets/icons/common/favorite.svg"></span>
    </div>
    <div class="row-metadataContainer">
      <div class="row-metadata"><img svg-inline="" src="../../assets/icons/datasets/tag.svg" class="icon-metadata">
        <ul class="tag-list">
          <li><a href="#" class="text is-small is-txtSoftGrey">Hi</a><span class="text is-small is-txtSoftGrey">,&nbsp;</span></li>
          <li><a href="#" class="text is-small is-txtSoftGrey">Hello</a><span class="text is-small is-txtSoftGrey">,&nbsp;</span></li>
          <li><a href="#" class="text is-small is-txtSoftGrey">CARTO Rules</a>
            <!---->
          </li>
        </ul>
        <!---->
      </div>
      <!---->
    </div>
  </div>
  <div class="dataset-cell cell--large"><span class="text is-small is-txtSoftGrey">DatasetCard.lastUpdated</span></div>
  <div class="dataset-cell cell--small"><span class="text is-small is-txtSoftGrey">DatasetCard.numberRows</span></div>
  <div class="dataset-cell cell--small"><span class="text is-small is-txtSoftGrey">34.4 MB</span></div>
  <div class="dataset-cell cell--small"><span class="text is-small is-txtSoftGrey">1 map</span></div>
  <div class="dataset-cell cell--small cell--privacy"><span class="icon icon--privacy icon--private"></span> <span class="text is-small is-txtSoftGrey">DatasetCard.shared.PRIVATE</span></div>
  <div class="dataset-cell"></div>
</a>
`;

exports[`DatasetCard.vue should select the card when the checkbox is clicked 1`] = `
<a href="http://example.com/tables/untitled_table_13" class="dataset-row selected">
  <div class="dataset-cell cell--start">
    <div class="row-dataType">
      <div class="icon--dataType icon--point"></div>
    </div> <span class="checkbox row-checkbox"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../../assets/icons/common/checkbox.svg"></span></span>
  </div>
  <div class="dataset-cell cell--main">
    <div class="title-container">
      <h3 class="text is-caption is-txtGrey u-ellipsis row-title">
        untitled_table_13
      </h3> <span class="card-favorite"><img svg-inline="" src="../../assets/icons/common/favorite.svg"></span>
    </div>
    <!---->
  </div>
  <div class="dataset-cell cell--large"><span class="text is-small is-txtSoftGrey">DatasetCard.lastUpdated</span></div>
  <div class="dataset-cell cell--small"><span class="text is-small is-txtSoftGrey">DatasetCard.numberRows</span></div>
  <div class="dataset-cell cell--small"><span class="text is-small is-txtSoftGrey">34.4 MB</span></div>
  <div class="dataset-cell cell--small"><span class="text is-small is-txtSoftGrey">1 map</span></div>
  <div class="dataset-cell cell--small cell--privacy"><span class="icon icon--privacy icon--private"></span> <span class="text is-small is-txtSoftGrey">DatasetCard.shared.PRIVATE</span></div>
  <div class="dataset-cell"></div>
</a>
`;

exports[`DatasetCard.vue should show the owner of the map if it is shared 1`] = `
<a href="http://example.com/tables/untitled_table_13" class="dataset-row">
  <div class="dataset-cell cell--start">
    <div class="row-dataType">
      <div class="icon--dataType icon--point"></div>
    </div> <span class="checkbox row-checkbox"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../../assets/icons/common/checkbox.svg"></span></span>
  </div>
  <div class="dataset-cell cell--main">
    <div class="title-container">
      <h3 class="text is-caption is-txtGrey u-ellipsis row-title">
        untitled_table_13
      </h3> <span class="card-favorite"><img svg-inline="" src="../../assets/icons/common/favorite.svg"></span>
    </div>
    <div class="row-metadataContainer">
      <!---->
      <div class="row-metadata"><img svg-inline="" src="../../assets/icons/datasets/user.svg" class="icon-metadata"> <span class="text is-small is-txtSoftGrey">cdb</span></div>
    </div>
  </div>
  <div class="dataset-cell cell--large"><span class="text is-small is-txtSoftGrey">DatasetCard.lastUpdated</span></div>
  <div class="dataset-cell cell--small"><span class="text is-small is-txtSoftGrey">DatasetCard.numberRows</span></div>
  <div class="dataset-cell cell--small"><span class="text is-small is-txtSoftGrey">34.4 MB</span></div>
  <div class="dataset-cell cell--small"><span class="text is-small is-txtSoftGrey">1 map</span></div>
  <div class="dataset-cell cell--small cell--privacy"><span class="icon icon--privacy icon--private"></span> <span class="text is-small is-txtSoftGrey">DatasetCard.shared.PRIVATE</span></div>
  <div class="dataset-cell"></div>
</a>
`;
